bundles/org.dataflowanalysis.converter/src/org/dataflowanalysis/converter/PCMConverter.java

```diff
@@ -342,21 +428,424 @@ private Label getOrCreateDFDLabel(CharacteristicValue charValue) {
+        Assignment assignment = datadictionaryFactory.eINSTANCE.createAssignment();
+    	LabelType labelType = getOrCreateLabelType(characteristicType.getName());
+
+        Label label = getOrCreateDFDLabel(characteristicValue.getName(), labelType);
+        assignment.getOutputLabels().add(label);
+
+        Pin outPin = node.getBehaviour().getOutPin().stream()
+                .filter(it -> it.getEntityName().equals(reference.getReferenceName()))
+                .findAny().orElseThrow();
+        assignment.setOutputPin(outPin);
+        org.dataflowanalysis.dfd.datadictionary.Term term = parseTerm(rightHandSide, dataDictionary);
+        assignment.setTerm(term);
+        if (rightHandSide instanceof NamedEnumCharacteristicReference namedEnumCharacteristicReference) {
+            Pin inPin = node.getBehaviour().getInPin().stream()
+                    .filter(it -> it.getEntityName().equals(namedEnumCharacteristicReference.getNamedReference().getReferenceName()))
+                    .findAny().orElseThrow();
+            assignment.getInputPins().add(inPin);
+        } else {
+        	assignment.getInputPins().addAll(behaviour.getInPin());
+        }
+        return assignment;
+    }
+    
+    /**
+     * Converts behavior from PCM Vertex to Node and adds it to the Data Dictionary
+     * @param pcmVertex the PCM Vertex
+     * @param node the DFD Node
+     * @param dataDictionary the Data Dictionary
+     */
+    public void convertBehavior(AbstractPCMVertex<?> pcmVertex, Node node, DataDictionary dataDictionary) {
+    	List<ConfidentialityVariableCharacterisation> variableCharacterisations = this.getVariableCharacterizations(pcmVertex);
 
+        Behaviour behaviour = node.getBehaviour();
+        List<AbstractAssignment> assignments = this.getAssignments(pcmVertex, node);
+        for (ConfidentialityVariableCharacterisation variableCharacterization : variableCharacterisations) {
+        	assignments.addAll(this.processCharacterization(variableCharacterization, behaviour, node, pcmVertex));
+        }
+        behaviour.getAssignment().clear();
+        behaviour.getAssignment().addAll(assignments);
+
+        behaviour.getOutPin().stream().forEach(pin -> {
+        	Assignment assignment = datadictionaryFactory.eINSTANCE.createAssignment();
+        	assignment.setEntityName("Dummy Assignment");
+        	assignment.getInputPins().addAll(behaviour.getInPin());
+        	assignment.setOutputPin(pin);
+        	assignment.setTerm(datadictionaryFactory.eINSTANCE.createTRUE());
+        	behaviour.getAssignment().add(assignment);
+        });
+    }
+
+    /**
+     * Parses a Term object into a corresponding Data Dictionary Term
+     * @param rightHandSide the Term object to parse
+     * @param dataDictionary the Data Dictionary
+     * @return the parsed Term
+     */
+    public org.dataflowanalysis.dfd.datadictionary.Term parseTerm(Term rightHandSide, DataDictionary dataDictionary) {
```

Comment:
This does lack some sort of recursion and assumes there is only one level, ie. A and B
Stuff like (A or B) and C would not be correctly parsed.

Please add testing for complex term parsing

Reply:
Incorrect, the `And` and `Or` cases both call `parseTerm` with both sides of the expression.
Tests however could be added

Reply:
My bad, missed the recursion

Reply:
Fixed in 7274f023380a